src/mod_keystore.erl

```diff
@@ -79,6 +82,33 @@ stop(Domain) ->
     clear_keystore_ets(Domain),
     ok.
 
+-spec config_spec() -> mongoose_config_spec:config_section().
+config_spec() ->
+    #section{
+        items = #{<<"ram_key_size">> => #option{type = integer,
+                                                validate = non_negative},
+                  <<"keys">> => #list{items = keys_spec()}
+        }
+    }.
+
+keys_spec() ->
+    #section{
+        items = #{<<"name">> => #option{type = atom,
+                                        validate = non_empty},
+                  <<"type">> => #option{type = atom,
+                                        validate = {enum, [file, ram]}},
+                  <<"path">> => #option{type = string,
+                                        validate = filename}
+        },
+        required = [<<"name">>, <<"type">>],
+        process = fun ?MODULE:process_keys/1
+    }.
+
+process_keys([{name, Name},{type, ram}]) ->
+    {Name, ram};
+process_keys([{name, Name}, {path, File}, {type, file}]) ->
+    {Name, {file, File}}.
+
```

Comment:
We should not rely on order for KV-lists, please use the `proplists` module instead.